pilot/pkg/kube/inject/inject.go

```diff
@@ -229,25 +230,26 @@ func (p *Params) Validate() error {
 }
 
 // intoHelmValues returns a map of the traversed path in helm values YAML to the param value.
-func (p *Params) intoHelmValues() map[string]string {
-	vals := map[string]string{
+func (p *Params) intoHelmValues() map[string]interface{} {
+	vals := map[string]interface{}{
 		"global.proxy_init.image":                    p.InitImage,
 		"global.proxy.image":                         p.ProxyImage,
-		"global.proxy.enableCoreDump":                strconv.FormatBool(p.EnableCoreDump),
-		"global.proxy.privileged":                    strconv.FormatBool(p.Privileged),
+		"global.proxy.enableCoreDump":                p.EnableCoreDump,
+		"global.proxy.privileged":                    p.Privileged,
 		"global.imagePullPolicy":                     p.ImagePullPolicy,
-		"global.proxy.statusPort":                    strconv.Itoa(p.StatusPort),
+		"global.proxy.statusPort":                    p.StatusPort,
 		"global.proxy.tracer":                        p.Tracer,
-		"global.proxy.readinessInitialDelaySeconds":  strconv.Itoa(int(p.ReadinessInitialDelaySeconds)),
-		"global.proxy.readinessPeriodSeconds":        strconv.Itoa(int(p.ReadinessPeriodSeconds)),
-		"global.proxy.readinessFailureThreshold":     strconv.Itoa(int(p.ReadinessFailureThreshold)),
-		"global.sds.enabled":                         strconv.FormatBool(p.SDSEnabled),
-		"global.sds.useTrustworthyJwt":               strconv.FormatBool(p.EnableSdsTokenMount),
+		"global.proxy.readinessInitialDelaySeconds":  p.ReadinessInitialDelaySeconds,
+		"global.proxy.readinessPeriodSeconds":        p.ReadinessPeriodSeconds,
+		"global.proxy.readinessFailureThreshold":     p.ReadinessFailureThreshold,
+		"global.sds.enabled":                         p.SDSEnabled,
+		"global.sds.useTrustworthyJwt":               p.EnableSdsTokenMount,
 		"global.proxy.includeIPRanges":               p.IncludeIPRanges,
 		"global.proxy.excludeIPRanges":               p.ExcludeIPRanges,
 		"global.proxy.includeInboundPorts":           p.IncludeInboundPorts,
 		"global.proxy.excludeInboundPorts":           p.ExcludeInboundPorts,
-		"sidecarInjectorWebhook.rewriteAppHTTPProbe": strconv.FormatBool(p.RewriteAppHTTPProbe),
+		"sidecarInjectorWebhook.rewriteAppHTTPProbe": p.RewriteAppHTTPProbe,
+		"global.podDNSSearchNamespaces":              p.PodDNSSearchNamespaces,
```

Comment:
I think these should still be strings, since this is testing Helm which takes in the values as strings. You can pass an array to helm like `--set foo={a,b,c}`